### PR TITLE
DEV-7 add GCP

### PR DESCRIPTION
**Pull request by daroczig**
There are still a few TODOs:
- [ ] bump version number
- [ ] changelog
- [ ] add logos
- [ ] update green_energy for 80% or 90% threshold instead of 100%
- [ ] double check on the `n1` to `m1` conversion .. all instances?

But the main part is ready for review :pray:

### CODE REVIEW

File: src/sc_crawler/vendors/gcp.py [hunk 1]
```diff
@@ -1,38 +1,875 @@
...
+            elif "Extreme PD Capacity" in sku.description:
+                storage_name = "pd-extreme"
+            elif "Hyperdisk Extreme Capacity" in sku.description:
+                storage_name = "hyperdisk-extreme"
+            elif "Hyperdisk Throughput Capacity" in sku.description:
+                storage_name = "hyperdisk-throughput"
+            elif "Hyperdisk Throughput Capacity" in sku.description:
+                storage_name = "hyperdisk-balanced"
+            else:
+                continue
```
Comment (bra-fsn) [inline src/sc_crawler/vendors/gcp.py:221]: I would replace this with sth like:
```suggestion
        for k in STORAGE_MAP.keys():
            if k in desc:
                storage_name = STORAGE_MAP[k]
                break
        else:
            continue
```
and add a constant to the top:
```python
STORAGE_MAP = {
    "Storage PD Capacity": "pd-standard",
    "SSD backed PD Capacity": "pd-ssd",
    "SSD backed Local Storage": "local-ssd",
    "Balanced PD Capacity": "pd-balanced",
}
```

**Review by bra-fsn — APPROVED** [review]

File: src/sc_crawler/lookup.py [hunk 2]
```diff
@@ -7,10 +7,12 @@
 country_continent_mapping = {
     "AE": "Asia",
     "AU": "Oceania",
+    "BE": "Europe",
```
[the remark this answers is not in the document]
Comment (daroczig) [reply src/sc_crawler/lookup.py:10]: that's where I started, but after wasting hours on selecting the right package, and then integrating and running into bugs, [decided to drop](https://github.com/SpareCores/sc-crawler/commit/6098fd9b707aca09e7a71af8dcb18c4844f3a6df) as all we need is a lookup for the continent -- seemed like easier to maintain :)

**Comment by daroczig** [comment]
thanks a lot for the useful feedback :heart_eyes_cat:

File: src/sc_crawler/vendors/gcp.py [hunk 3]
```diff
@@ -1,38 +1,875 @@
...
+    Some data sources are not available from APIs, and were collected manually:
+    - location: <https://cloud.google.com/compute/docs/regions-zones#available> and <https://en.wikipedia.org/wiki/Google_data_centers>
+    - energy carbon data: <https://cloud.google.com/sustainability/region-carbon#data> and <https://github.com/GoogleCloudPlatform/region-carbon-info>
+    - launch dates were collected from [Wikipedia](https://en.wikipedia.org/wiki/Google_Cloud_Platform#Regions_and_zones) and GCP blog posts, such as <https://medium.com/@retomeier/an-annotated-history-of-googles-cloud-platform-90b90f948920> and <https://cloud.google.com/blog/products/infrastructure/introducing-new-google-cloud-regions>
+
+    Note that many GCP datacenters use more than 90% green energy,
+    but the related flag in our database is set to `False` as not being 100%.
+    """
+
+    manual_data = {
```
[the remark this answers is not in the document]
Comment (daroczig) [reply src/sc_crawler/vendors/gcp.py:348]: The advantage of having this as code in a long format is that we can add comments on each value, e.g., the source of that info, whether something needs to be verified, etc. With CSV, that would only be possible row-wise in an extra column :zipper_mouth_face: 

If there's a strong preference, I'm open to moving this mapping out of this file or even into another format, but I'd prefer keeping the option to comment on each value.

File: src/sc_crawler/vendors/gcp.py [hunk 4]
```diff
@@ -1,38 +1,883 @@
...
+                family = sub(r"^Spot Preemptible ", "", sku.description)
+                family = sub(r" Instance.*", "", family)
+                family = sub(r" AMD$", "", family)
+                family = sub(r" Arm$", "", family)
+
+                # extract instance family from description (?!)
+                for k, v in SERVER_DESCRIPTION_TO_FAMILY.items():
+                    if family == k:
+                        family = v
+                family = family.lower()
```
Comment (bra-fsn) [inline src/sc_crawler/vendors/gcp.py:215]: ... wouldn't be this enough here?
```suggestion
                family = SERVER_DESCRIPTION_TO_FAMILY.get(family, family).lower()
```
I'm not sure if a miss could be possible, but the above code should replicate the same even in that case.

**Pull request merged**